### ISSUE TITLE
I would like to contribute with this development

### ISSUE THREAD

**Issue by Trololui** [UltimateFishbot/UltimateFishbot#97]
Hi there,

I would like to have a conversation with one of the contributor just to get an overview of the project, I checked the code and as always there is some room for improvement.

@ryanrhee could you tell me what is the purpose of this:

>                 // Remove the extras spaces from each lines
>                 returnList.Select(text => String.Join("\n", text.Split('\n').Select(s => s.Trim())));
> 
>                 // Replace {int} in text by variables. Ex : "Waiting for Fish ({0}/{1}s) ..."
>                 returnList.Select(text => String.Format(text, list));

The .Select is not doing a thing here as it doesn't change the object. I'm trying to get the idea and I guess what you are trying to do is to remove the extra spaces and from the words and then create a single string, right?

It seems like we can solve this by making sure the translations are well formatted, so this ends up being unnecessary, yes we can put some defensive code but that's it.

Am I wrong?

The "Costura.Fody" package, how are you using it? I can only see an empty xml file.

Once again I'm more than happy to help with I can but I need more info about the expected workflow of the project.

I need some help to help the project.

@UltimateFishbot I hope this is the right place to ask.

Thanks

**Comment by ryanrhee** [comment]
@Trololui does it compile without the .Select ?

Also, can you be more specific? Which one are you referring to? In the example, there are 3 distinct .Select() calls. Do you mean to point out one specific instance of it, or all of them?

**Comment by ryanrhee** [comment]
@Trololui 

> The "Costura.Fody" package, how are you using it?

I'm not sure about that one. It might be for translations or for text-to-speech? One of the original contributors is likely better suited to answer that question. My contributions to this project invovled re-writing the core to use async/await/tasks. At least in that part, I don't think I needed the "Costura.Fody" package.

But it might be easier for me to help you if you can tell me exactly what you want to change. Is there a specific bug or feature you want to work on, or are you looking to improve the overall code quality of the repo?

**Comment by Trololui** [comment]
@ryanrhee I would like to first improve the overall code quality of the repo when that is done I'll probably have a better idea of how things work, it's good to mention that is the first time I do a wow addon thingy.

The "Costura.Fody" package is not being used as far as I can see.

Regarding the .Select I'm talking about all of them, in order to use the .Select you need to create an object to assign the result of it. e.g.

`var t = returnList.Select(text => String.Join("\n", text.Split('\n').Select(s => s.Trim())));`

so t will have the new result.

Basically, those 2 lines are doing nothing.

I'll start refactoring what I think I can and with a few PR's you guys will see if the direction I'm going is something we all like.

**Comment by ryanrhee** [comment]
@Trololui the .Select() calls you mentioned are all in the Translate.cs file, right?

**Comment by NaNraptor** [comment]
Closing due to no activity
